Add setting to change the default cache used

induane opened the pull request:
It is handy to be able to flush the ldap cache without flushing the entire default django cache. That can reset non-token based logins and all kinds of data you may or may not want. Being able to configure it to use it's own independent cache within the Django caching framework is reallllly handy!

I had previously submitted this before but the tree was really out of date so it was easier to incorporate the suggested changes in a new branch and resubmit.

francoisfreitag commented on docs/reference.rst:
```diff
@@ -341,6 +341,33 @@ Values may be strings for simple group membership tests or
+AUTH_LDAP_CACHE
+~~~~~~~~~~~~~~~
+
+Default: ``default``
```
Add quotes to indicate it’s a string.
```suggestion
Default: ``'default'``
```

francoisfreitag reviewed (commented):
From #199

Nice improvement. Can you please add a test?

Here are the suggestions from the last PR:

1. trigger an operation that creates a cache entry
2. verify:
    - that the cache entry was created in the configured cache, and
    - that the cache entry does not exist Django default cache

induane commented:
Thanks for the feedback everyone. I've added the unittest and updated the documentation as suggested. Let me know if there is anything further I need to do.
